src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -236,4 +237,42 @@ private function getRelationDataFromFormData($data)
 
         return $relationData;
     }
+
+    /**
+     * Transform the data before saving into the database. This is kind of an alias to Laravel Mutators but defined on Controler when defining the field.
+     * We basically call this "mutator" before saving the data into the database but after data is already validated.
+     *
+     * @param  array  $data  The form data.
+     * @return array The processed data.
+     */
+    protected function handleBeforeSavingCallback($data)
+    {
+        $fields = array_filter($this->fields(), function ($field) use ($data) {
```

Comment:
```suggestion
        $fieldsWithBeforeSaving = array_filter($this->fields(), function ($field) use ($data) {
```

Let's better name the variables. These are not just fields, but fields with before saving defined. The code would be easier to read if the variable name let you know that.

Reply:
I agree with your reasoning, but I also think that in this scenario `$fields` is a scoped variable, in this scenario I think it's implicit on the definition of the variable what fields they are. Anyway, done 😃